boards/shields/nrf2240ek/src/shield.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+
+#define NPM1300_REGADDR_VBUSIN_TASKUPDATEILIMSW 0x0200U
+
+#define NPM1300_REGVAL_VBUSIN_TASKUPDATEILIMSW_NOEFFECT     0x00U
+#define NPM1300_REGVAL_VBUSIN_TASKUPDATEILIMSW_SELVBUSILIM0 0x01U
+
+#define NPM1300_REGADDR_VBUSIN_VBUSINILIM0      0x0201U
+
+#define NPM1300_REGVAL_VBUSIN_VBUSINILIM0_500MA0 0U
+
+/**
+ * @brief Writes a register of a nPM1300 device through I2C.
+ *
+ * @param spec     I2C specification from the devicetree.
+ * @param reg_addr Address of the register within nPM1300 device.
+ * @param value    The value to write into the register.
+ *
+ * @return As underlying @ref i2c_write_dt function. @c 0 if successful.
+ */
+static inline int npm1300_reg_write_dt(
+		const struct i2c_dt_spec *spec,
+		uint16_t reg_addr,
+		uint8_t value)
+{
+	/* Note: There is only 8-bit reg_addr version in drivers/i2c.h provided by Zephyr.
+	 * The nPM1300 uses the 16-bit register address.
+	 */
+
+	uint8_t tx_buf[3] = {
+		(uint8_t)(reg_addr >> 8),
+		(uint8_t)reg_addr,
+		value
+	};
+
+	return i2c_write_dt(spec, tx_buf, 3);
+}
+
+static int nrf2240ek_pmic_init(void)
+{
+	int err;
+	struct i2c_dt_spec bus = I2C_DT_SPEC_GET(DT_NODELABEL(nrf2240ek_pmic));
+
```

Comment:
Rationale for direct i2c writes.

The `regulator.h` API promises the function `regulator_set_current_limit`
https://github.com/nrfconnect/sdk-zephyr/blob/fb6ca20a06ba31a3dc1b86d6986d48ad4a5760b0/include/zephyr/drivers/regulator.h#L604

 but it is not implemented
https://github.com/nrfconnect/sdk-zephyr/blob/fb6ca20a06ba31a3dc1b86d6986d48ad4a5760b0/drivers/regulator/regulator_npm1300.c#L637

Reply:
`regulator_set_current_limit` for VBUS is not implemented because USB interface of PMIC is not considered a regulator. BUCK1, BUCK2, LDO1 and LDO2 are regulators, and they indeed don't support setting of a current limit.

There is `charger` interface you could use, I'll post a more detailed answer below in a moment.